pkg/restore/restore.go

```diff
@@ -259,50 +259,98 @@ func (ctx *context) execute() (api.RestoreResult, api.RestoreResult) {
 func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreResult) {
 	warnings, errors := api.RestoreResult{}, api.RestoreResult{}
 
-	// cluster-scoped
-	clusterPath := path.Join(dir, api.ClusterScopedDir)
-	exists, err := ctx.fileSystem.DirExists(clusterPath)
-	if err != nil {
-		errors.Cluster = []string{err.Error()}
-	}
-	if exists {
-		w, e := ctx.restoreNamespace("", clusterPath)
-		merge(&warnings, &w)
-		merge(&errors, &e)
-	}
+	namespaceFilter := collections.NewIncludesExcludes().Includes(ctx.restore.Spec.IncludedNamespaces...).Excludes(ctx.restore.Spec.ExcludedNamespaces...)
 
-	// namespace-scoped
-	namespacesPath := path.Join(dir, api.NamespaceScopedDir)
-	exists, err = ctx.fileSystem.DirExists(namespacesPath)
+	// Make sure the top level "resources" dir exists:
+	resourcesDir := path.Join(dir, api.ResourcesDir)
+	_, err := ctx.fileSystem.DirExists(resourcesDir)
 	if err != nil {
 		addArkError(&errors, err)
 		return warnings, errors
 	}
-	if !exists {
-		return warnings, errors
-	}
 
-	nses, err := ctx.fileSystem.ReadDir(namespacesPath)
+	resourceDirs, err := ctx.fileSystem.ReadDir(resourcesDir)
 	if err != nil {
 		addArkError(&errors, err)
 		return warnings, errors
 	}
 
-	namespaceFilter := collections.NewIncludesExcludes().Includes(ctx.restore.Spec.IncludedNamespaces...).Excludes(ctx.restore.Spec.ExcludedNamespaces...)
-	for _, ns := range nses {
-		if !ns.IsDir() {
+	resourceDirsMap := make(map[string]os.FileInfo)
+
+	for _, rscDir := range resourceDirs {
+		rscName := rscDir.Name()
+		resourceDirsMap[rscName] = rscDir
+	}
+
+	for _, resource := range ctx.prioritizedResources {
+		rscDir := resourceDirsMap[resource.String()]
+		if rscDir == nil {
 			continue
 		}
-		nsPath := path.Join(namespacesPath, ns.Name())
 
-		if !namespaceFilter.ShouldInclude(ns.Name()) {
-			ctx.infof("Skipping namespace %s", ns.Name())
-			continue
+		resourcePath := path.Join(resourcesDir, rscDir.Name())
+
+		clusterSubDir := path.Join(resourcePath, api.ClusterScopedDir)
+		clusterSubDirExists, err := ctx.fileSystem.DirExists(clusterSubDir)
+		if err != nil {
+			addArkError(&errors, err)
+			return warnings, errors
+		}
+		if clusterSubDirExists {
+			w, e := ctx.restoreResource(resource.String(), "", clusterSubDir)
+			merge(&warnings, &w)
+			merge(&errors, &e)
```

Comment:
`continue`?

Reply:
Must be no such thing as a resource that could be cluster or namespace scoped. Will add that in.